### PR TITLE
[search] minimal dashboard indexing

### PR DESCRIPTION
**Pull request by scottlepp**
index a dashboard resource with minimal field mapping ( more fields need to be mapped )

### CODE REVIEW

File: pkg/storage/unified/resource/index_test.go [hunk 1]
```diff
@@ -110,3 +103,38 @@ func namespaces() []string {
...
+			BatchSize: 1000,
+		},
+	}
+}
+
+func assertNil(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
```
Comment (JohnnyQQQQ) [inline pkg/storage/unified/resource/index_test.go:128]: Is there as specific reason why you wouldn't use `require.NoError` instead?

File: pkg/storage/unified/resource/index_test.go [hunk 2]
```diff
@@ -110,3 +103,38 @@ func namespaces() []string {
...
+
+func assertCount(t *testing.T, index *Index, expected uint64) {
+	total, err := index.Count()
+	assertNil(t, err)
+	assert.Equal(t, expected, total)
```
Comment (JohnnyQQQQ) [inline pkg/storage/unified/resource/index_test.go:133]: I wouldn't put this into a dedicated function, it's only used two times. Even if used multiple times, having it directly in the testing code makes it much easier to understand the code.

I would also change the API to something like that:

```golang
total, err := index.Count()
require.NoError(t, err)
require.Equal(t, expected, total)
```

Comment (scottlepp) [reply pkg/storage/unified/resource/index_test.go:128]: it's only used 2 times but we will be adding many more test scenarios

**Review by JohnnyQQQQ — COMMENTED** [review]
LGTM, really cool to see this coming alive! Will let someone from the team make the final approval.

**Review by owensmallwood — APPROVED** [review]
lgtm 🚀

**Pull request merged**